# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/media/webrtc/MediaEngineRemoteVideoSource.cpp
# Commit: 20543d03a4cd
# Full Hash: 20543d03a4cda05c785011a25079aa42aec8b2ba
# Author: Andreas Pehrson <apehrson@mozilla.com>
# Date: 2019-03-29 21:53:01
# Regressor Bug: 1494675
# File Overlap Count: 2
# Description:
#   Bug 1494675 - Remove AllocationHandle. r=padenot
#   
#   The handle was used to keep separate allocations of the same source in a single
#   process apart. Sources no longer use sharing, so we no longer need allocations
#   or their handles even as a concept.
# ==============================================================================

diff -r 9e7832fee028 -r 20543d03a4cd dom/media/webrtc/MediaEngineRemoteVideoSource.cpp
--- a/dom/media/webrtc/MediaEngineRemoteVideoSource.cpp	Wed Mar 27 14:10:27 2019 +0000
+++ b/dom/media/webrtc/MediaEngineRemoteVideoSource.cpp	Wed Mar 27 14:10:24 2019 +0000
@@ -5,7 +5,6 @@
 
 #include "MediaEngineRemoteVideoSource.h"
 
-#include "AllocationHandle.h"
 #include "CamerasChild.h"
 #include "MediaManager.h"
 #include "MediaTrackConstraints.h"
@@ -100,13 +99,11 @@
     return;
   }
 
-  // Allocate always returns a null AllocationHandle.
-  // We can safely pass nullptr here.
   if (mState == kStarted) {
-    Stop(nullptr);
+    Stop();
   }
   if (mState == kAllocated || mState == kStopped) {
-    Deallocate(nullptr);
+    Deallocate();
   }
   MOZ_ASSERT(mState == kReleased);
 
@@ -197,7 +194,7 @@
     const MediaTrackConstraints& aConstraints, const MediaEnginePrefs& aPrefs,
     const nsString& aDeviceId,
     const mozilla::ipc::PrincipalInfo& aPrincipalInfo,
-    AllocationHandle** aOutHandle, const char** aOutBadConstraint) {
+    const char** aOutBadConstraint) {
   LOG(__PRETTY_FUNCTION__);
   AssertIsOnOwningThread();
 
@@ -225,8 +222,6 @@
     return NS_ERROR_FAILURE;
   }
 
-  *aOutHandle = nullptr;
-
   {
     MutexAutoLock lock(mMutex);
     mState = kAllocated;
@@ -237,8 +232,7 @@
   return NS_OK;
 }
 
-nsresult MediaEngineRemoteVideoSource::Deallocate(
-    const RefPtr<const AllocationHandle>& aHandle) {
+nsresult MediaEngineRemoteVideoSource::Deallocate() {
   LOG(__PRETTY_FUNCTION__);
   AssertIsOnOwningThread();
 
@@ -273,7 +267,6 @@
 }
 
 void MediaEngineRemoteVideoSource::SetTrack(
-    const RefPtr<const AllocationHandle>& aHandle,
     const RefPtr<SourceMediaStream>& aStream, TrackID aTrackID,
     const PrincipalHandle& aPrincipal) {
   LOG(__PRETTY_FUNCTION__);
@@ -300,8 +293,7 @@
                     SourceMediaStream::ADDTRACK_QUEUED);
 }
 
-nsresult MediaEngineRemoteVideoSource::Start(
-    const RefPtr<const AllocationHandle>& aHandle) {
+nsresult MediaEngineRemoteVideoSource::Start() {
   LOG(__PRETTY_FUNCTION__);
   AssertIsOnOwningThread();
 
@@ -355,8 +347,7 @@
   return NS_OK;
 }
 
-nsresult MediaEngineRemoteVideoSource::FocusOnSelectedSource(
-    const RefPtr<const AllocationHandle>& aHandle) {
+nsresult MediaEngineRemoteVideoSource::FocusOnSelectedSource() {
   LOG(__PRETTY_FUNCTION__);
   AssertIsOnOwningThread();
 
@@ -366,8 +357,7 @@
   return result == 0 ? NS_OK : NS_ERROR_FAILURE;
 }
 
-nsresult MediaEngineRemoteVideoSource::Stop(
-    const RefPtr<const AllocationHandle>& aHandle) {
+nsresult MediaEngineRemoteVideoSource::Stop() {
   LOG(__PRETTY_FUNCTION__);
   AssertIsOnOwningThread();
 
@@ -392,7 +382,6 @@
 }
 
 nsresult MediaEngineRemoteVideoSource::Reconfigure(
-    const RefPtr<AllocationHandle>& aHandle,
     const MediaTrackConstraints& aConstraints, const MediaEnginePrefs& aPrefs,
     const nsString& aDeviceId, const char** aOutBadConstraint) {
   LOG(__PRETTY_FUNCTION__);
@@ -417,9 +406,7 @@
 
   bool started = mState == kStarted;
   if (started) {
-    // Allocate always returns a null AllocationHandle.
-    // We can safely pass nullptr below.
-    nsresult rv = Stop(nullptr);
+    nsresult rv = Stop();
     if (NS_WARN_IF(NS_FAILED(rv))) {
       nsAutoCString name;
       GetErrorName(rv, name);
@@ -437,7 +424,7 @@
   }
 
   if (started) {
-    nsresult rv = Start(nullptr);
+    nsresult rv = Start();
     if (NS_WARN_IF(NS_FAILED(rv))) {
       nsAutoCString name;
       GetErrorName(rv, name);